packages/technical-features/feature-core/src/register-feature.ts

```diff
@@ -0,0 +1,90 @@
+import type { DiContainer } from "@ogre-tools/injectable";
+import { getInjectable } from "@ogre-tools/injectable";
+import type { Feature } from "./feature";
+import {
+  featureContextMapInjectable,
+  featureContextMapInjectionToken,
+} from "./feature-context-map-injectable";
+
+export const registerFeature = (di: DiContainer, ...features: Feature[]) => {
+  features.forEach((feature) => {
+    registerFeatureRecursed(di, feature);
+  });
+};
+
+const registerFeatureRecursed = (
+  di: DiContainer,
+  feature: Feature,
+  dependedBy?: Feature
+) => {
```

Comment:
_nit_: this is more readable imho + there is no actual reason to use arrow-function, `this` context is not even considered).

```suggestion
function registerFeatureRecursed(di: DiContainer, feature: Feature, dependedBy?: Feature) {
```